src/database/MySQLTable.ts

```diff
@@ -26,7 +26,7 @@ export class MySQLTable {
     this.buffer[row.action].push(row);
 
     if (this.buffer[row.action].length >= this.flushLimit) {
-      return this.flush(row.action);
+      return await this.flush(row.action);
```

Comment:
Was this omission intentional?

Reply:
Adding await there doesn't do anything unless it's wrapped in a try catch statement. Await just yields a promise, so there you are yielding, then continuing once the promise is resolved and returning the value. Returning the promise of something is the same

Reply:
Without it the code that called it returned immediately. With it it waited for the flush to be complete. I could not get it to work correctly without the await here.